sample/SampleHost/Program.cs

```diff
@@ -20,6 +23,23 @@ static void Main(string[] args)
             }
 
             config.CreateMetadataProvider().DebugDumpGraph(Console.Out);
+            config.Tracing.ConsoleLevel = TraceLevel.Off;
+
+            // Build up a LoggerFactory to log to App Insights, but only if this key exists.
+            string instrumentationKey = Environment.GetEnvironmentVariable("APPINSIGHTS_INSTRUMENTATIONKEY");
+            if (!string.IsNullOrEmpty(instrumentationKey))
+            {
+                // build up log levels for any category
+                LogCategoryFilter filterBuilder = new LogCategoryFilter();
```

Comment:
It used to be called FilterBuilder in a previous iteration. Someone recommended I rename it. I can change that local.

I want to keep it generic enough so that everyone doesn't have to build their own Func from scratch. If you look at other Loggers, they all take `Func<string, LogLevel, bool>`. You can see that here, where I pass the Filter in. It makes it usable in other standard loggers.

Some examples: 
- https://github.com/aspnet/Logging/blob/dev/src/Microsoft.Extensions.Logging.Console/ConsoleLogger.cs#L33
- https://github.com/aspnet/Logging/blob/dev/src/Microsoft.Extensions.Logging.Debug/DebugLogger.cs#L31